sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md

```diff
@@ -0,0 +1,97 @@
+
+## Testing project structure
+
+With the help of [Azure.ResourceManager.Template][mgmt_template], the basic testing project and file structure(see following for details) will be generated under `sdk\<service name>\<package name>\tests` directory.
+
+```text
+sdk\<service name>\<package name>\tests\Azure.ResourceManager.<service>.Tests.csproj
+sdk\<service name>\<package name>\tests\<service>ManagementTestBase.cs
+sdk\<service name>\<package name>\tests\<service>ManagementTestEnvironment.cs
+sdk\<service name>\<package name>\tests\Scenario
+sdk\<service name>\<package name>\tests\SessionRecords
+```
+
+**Note**: Considering that in Git directories exist implicitly, so you might need to create the `Scenario` and `SessionRecords` directories by yourself after cloning the repo.
+
+## Writing scenario tests
+
+All the public APIs that are exposed to the customer need to be tested and they are distributed in the following three kinds of code files as shown below. Accordingly, you'd better put the tests of the same file's APIs in a separate file under the `Scenario` folder, thus facilitating the subsequent maintenance.
+
+```text
+sdk\<service name>\<package name>\src\Generated\Extensions\<service>Extensions.cs
+sdk\<service name>\<package name>\src\Generated\<resource>Collection.cs
+sdk\<service name>\<package name>\src\Generated\<resource>Resource.cs
+```
+
+For instance, if you want to test the `CreateOrUpdate` method in ExampleCollection.cs, the corresponding test file will be like:
+
+```csharp
+namespace Azure.ResourceManager.Service.Tests
+{
+    public class ExampleCollectionTests : ServiceManagementTestBase
+    {
+        public ExampleCollectionTests(bool isAsync)
+            : base(isAsync)//, RecordedTestMode.Record)
+        {
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task CreateOrUpdate()
+        {
+            SubscriptionResource subscription = await Client.GetDefaultSubscriptionAsync();
+            ResourceGroupResource rg = await CreateResourceGroup(subscription, "testRg", AzureLocation.WestUS);
+            string resourceName = Recording.GenerateAssetName("resource");
+            ExampleResource resource = await rg.GetExamples().CreateOrUpdateAsync(WaitUntil.Completed, resourceName, new ExampleData());
+            Assert.AreEqual(resourceName, resource.Data.Name);
+        }
+    }
+}
+```
+
+We expose all of our APIs with both sync and async variants. To avoid writing each of our tests twice, we automatically convert our async API calls into their sync equivalents at runtime. Simply write your tests using only async APIs and the `Azure.Core.TestFramework` will wrap the client with a proxy that forwards everything to the sync overloads automatically. Visual Studio's test runner will show `*TestClass(True)` for the async variants and `*TestClass(False)` for the sync variants.
```

Comment:
```suggestion
We expose all of our APIs with both sync and async variants. To avoid writing each of our tests twice, the test framework automatically converts async API calls into their sync equivalents at runtime. Simply write your tests using only async APIs and the `Azure.Core.TestFramework` will wrap the client with a proxy that forwards everything to the sync overloads automatically. Visual Studio's test runner will show `*TestClass(True)` for the async variants and `*TestClass(False)` for the sync variants.
```